integ-test/src/test/java/org/opensearch/sql/ppl/DataTypeIT.java

```diff
@@ -72,4 +74,23 @@ public void test_long_integer_data_type() throws IOException {
         schema("long2", "long"));
   }
 
+  @Test
+  public void typeof() throws IOException {
+    JSONObject response = executeQuery(String.format("source=%s | eval "
+        + "`str` = typeof('pewpew'), `null` = typeof(1/0), `double` = typeof(1.0),"
+        + "`int` = typeof(12345), `long` = typeof(1234567891011)"
+        + " | fields `str`, `null`, `double`, `int`, `long`",
+            TEST_INDEX_DATATYPE_NUMERIC));
+    verifyDataRows(response,
+        rows("STRING", "UNDEFINED", "DOUBLE", "INTEGER", "LONG"));
+
+    response = executeQuery(String.format("source=%s | eval "
+        + "`datetime` = typeof(CAST('1961-04-12 09:07:00' AS DATETIME)),"
+        + "`timestamp` = typeof(CAST('1961-04-12 09:07:00' AS TIMESTAMP)),"
+        + "`time` = typeof(CAST('09:07:00' AS TIME)), `date` = typeof(CAST('1961-04-12' AS DATE))"
+        + " | fields `datetime`, `timestamp`, `time`, `date`",
+            TEST_INDEX_DATATYPE_NUMERIC));
+    verifyDataRows(response,
+        rows("DATETIME", "TIMESTAMP", "TIME", "DATE"));
```

Comment:
please add a test for complex data types, like ARRAY, STRUCT, and INTERVAL.

Reply:
To test with `ARRAY` and `STRUCT` I have to update parser and `AstBuilder`/`AstExpressionBuilder`. I can't imagine where it could be used especially as a return value of another function.
`typeof(somefunc(...)) -> STRUCT`.